specs/Badge.md

```diff
@@ -0,0 +1,106 @@
+# Badge
+
+## Background
+
+A badge is an additional visual discriptor for UI elements.It can be used to denote numerical value, status or general information.
+
+## Prior Art
+
+_Include background research done for this component_
+
+- [OpenUI Research](https://open-ui.org/components/badge.research)
+- [AntD](https://ant.design/components/badge/)
+- [Atlassian](https://atlassian.design/components/badge/examples)
+- [Bootstrap](https://getbootstrap.com/docs/4.3/components/badge/)
+- [Fast](https://explore.fast.design/components/fast-badge)
+- [MaterialUI](https://material-ui.com/components/badges/#badge)
+- [Lightning](https://www.lightningdesignsystem.com/components/badges/)
+- [React Native](https://reactnativeelements.com/docs/badge/)
+
+## Sample Code
+
+```jsx
+  <Badge>
+    <Text>My Custom Badge</Text>
+  </Badge>
+  <Badge
+    status="success"
+    style={{ position: 'absolute', top: -4, right: -4 }}
+  />
+```
+
+## Variants
+
+`Badge` can have several variations.
+
+- Appearance: `default`, `rounded` and `circular`
+- Size: `smallest`, `smaller`, `small`, `medium`, `large`, `larger`.
+- Styles: `filled`, `outline`, `ghost`, `tint`, `inverted filled`
+
+## API
+
+| Property Name | Type      | Default Value | Description                  |
+| ------------- | --------- | ------------- | ---------------------------- |
+| `size`        | `enum`    | `medium`      | The Badge size               |
+| `status`      | `enum`    | `none`        | The Badge status color       |
+| `rounded`     | `boolean` | `false`       | If Badge has rounded corners |
+| `circular`    | `boolean` | `false`       | If Badge is circular         |
+| `filled`      | `boolean` | `true`        | If Badge is filled           |
+| `outline`     | `boolean` | `false`       | If Badge is outlined         |
+| `ghost`       | `boolean` | `false`       | If Badge is ghost            |
+| `inverted`    | `boolean` | `false`       | If Badge is inverted         |
+| `tint`        | `boolean` | `false`       | If badge is tint             |
```

Comment:
These boolean props (filled, outline, ghost, inverted, tint) are all mutually exclusive with each other. Perhaps they should be a single enum prop?

Reply:
Agreed

Reply:
I am simplifying the base badge to 

| Property Name | Type   | Default Value | Description                         |
| ------------- | ------ | ------------- | ----------------------------------- |
| `size`        | `enum` | `medium`      | The Badge size                      |
| `shape`       | `enum` | `rounded`     | The Badge can have different shapes |
| `variant`     | `enum` | `filled`      | THe Badge different style variants  |


Agree?